### PR TITLE
fix: clean SwapProps and standardize event handler logic

### PR DESCRIPTION
**Pull request by zzmp**

### CODE REVIEW

File: src/hooks/useConditionalHandler.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    async (...params) => {
+      if (!handler) return true
+
+      try {
+        const result = await handler(...params)
+        if (result === false) return false
+      } catch {
+        return false
+      }
+      return true
```
Comment (JFrankfurt) [inline src/hooks/useConditionalHandler.ts:16]: can replace these returns with a single `finally` block

**Review by JFrankfurt — APPROVED** [review]
one question and a nit--lgtm

File: src/components/Swap/index.tsx [hunk 2]
```diff
@@ -33,13 +33,7 @@ import useValidate from './useValidate'
 // since the wallet connection component exists within the Swap component.
 // This includes useSyncWidgetEventHandlers.
 // TODO(zzmp): refactor WalletConnection outside of Swap component
-export interface SwapProps
-  extends BrandingSettings,
-    FeeOptions,
-    SwapController,
```
[the remark this answers is not in the document]
Comment (zzmp) [reply src/components/Swap/index.tsx:39]: Yes, it should only need supplied through the 'value' prop now

File: src/hooks/useConditionalHandler.ts (hunk 1 above)
Comment (zzmp) [reply src/hooks/useConditionalHandler.ts:16]: I think a `finally` block will always run, so it overrides the returned value from within the try-catch block.

**Pull request merged**